### PR TITLE
20231011 cbmc 5.94.0

### PR DESCRIPTION
**Pull request by TGWDB**
PR for `cbmc` 5.94.0. Note that this is trialing including a `CHANGELOG` as discussed in #7907 .

The following have committed PRs and should review my editing of their PR title in the `CHANGELOG` to ensure I didn't mangle it: @thomasspriggs @peterschrammel @tautschnig @esteffin @NlightNFotis @qinheping @kroening 

- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [ ] My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- [ ] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: CHANGELOG
Comment (thomasspriggs) [inline CHANGELOG]: :pick: I don't think we have added enough support for any union examples to work from a user perspective.

File: CHANGELOG
Comment (thomasspriggs) [inline CHANGELOG]: :pick: I suspect this is actually a misleadingly named PR, as we enabled the tests which pass only. There are at least some remaining to fix.

**Review by thomasspriggs — APPROVED** [review]

File: CHANGELOG
Comment (tautschnig) [inline CHANGELOG]: I suggest not even including this section, though perhaps those could be listed in the PR description for reviewers to potentially override whoever is putting together the PR. My personal take is that the changelog should strongly focus on observable changes, where "observable" are all those changes that impact users in the first instance and developers if there are substantial API changes.

File: CHANGELOG
Comment (tautschnig) [inline CHANGELOG]: Not relevant to users.

File: CHANGELOG
Comment (tautschnig) [inline CHANGELOG]: I think those two items could be combined.

File: CHANGELOG
Comment (tautschnig) [inline CHANGELOG]: I'd reword this as "goto-synthesizer now accepts all CBMC options by @qinheping ..."

File: CHANGELOG
Comment (tautschnig) [inline CHANGELOG:3]: Before that line we'll need a `#`-level heading that states the version that these changes are referring to.

**Review by tautschnig — APPROVED** [review]
Thank you for introducing a changelog!

**Review by esteffin — APPROVED** [review]

**Review by NlightNFotis — APPROVED** [review]

**Comment by tautschnig** [comment]
Could we please get #7954 into this release (i.e., hold off releasing until this is approved, and then rebase this release)?

**Pull request merged**
